src/cache.js

```diff
@@ -19,7 +19,8 @@ class Cache {
       throw new Error('storage is required')
     }
 
-    if (options.ttl && (typeof options.ttl !== 'number' || options.ttl < 0 || !Number.isInteger(options.ttl))) {
+    // ttl _may_ be a function to defer the ttl decision until later
+    if (options.ttl && typeof options.ttl === 'number' && (options.ttl < 0 || !Number.isInteger(options.ttl))) {
```

Comment:
options.ttl _may_ be a function, so the logic around checking the numeric value being a positive integer needs to be guarded by a check that options.ttl is a number first.